### PR TITLE
CORS-3191: Add installer-etcd-artifacts image for etcd

### PR DESCRIPTION
**Pull request by r4f4**
This is opened to enable building the installer-etcd-artifacts image for etcd, being introduced in https://github.com/openshift/etcd/pull/249

### CODE REVIEW

**Review by dhellmann — COMMENTED** [review]
The image name looks OK. I'll leave the other changes up to your team to approve.

/approve

**Comment by r4f4** [comment]
Update: rebased to fix merge conflicts.

/pj-rehearse pull-ci-openshift-etcd-openshift-4.16-images

**Comment by r4f4** [comment]
@Elbehery I think we need https://github.com/openshift/etcd/pull/251 before this merges.

File: ci-operator/config/openshift/etcd/openshift-etcd-openshift-4.16.yaml [hunk 1]
```diff
@@ -32,6 +32,15 @@ images:
       as:
       - registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.20-openshift-4.16
   to: etcd
+- dockerfile_path: Dockerfile.installer
+  inputs:
+    ocp_4.16_base-rhel9:
+      as:
+      - registry.ci.openshift.org/ocp/4.16:base-rhel9
+    ocp_builder_rhel-9-etcd-golang-1.19:
```
Comment (Elbehery) [inline ci-operator/config/openshift/etcd/openshift-etcd-openshift-4.16.yaml:40]: @r4f4 i think there is a mistake 
```suggestion
    ocp_builder_rhel-9-etcd-golang-1.20:
```

Comment (r4f4) [reply ci-operator/config/openshift/etcd/openshift-etcd-openshift-4.16.yaml:40]: Yeah, the etcd build image was replaced by the rhel-9-golang-1.20-openshift-4.16 one. I'll update this once https://github.com/openshift/etcd/pull/251 merges.

**Comment by r4f4** [comment]
/pj-rehearse ci/rehearse/openshift/etcd/openshift-4.16/images

File: ci-operator/config/openshift/etcd/openshift-etcd-openshift-4.16.yaml (hunk 1 above)
Comment (r4f4) [reply ci-operator/config/openshift/etcd/openshift-etcd-openshift-4.16.yaml]: Fixed.

**Pull request merged**